x/incentives/types/incentive_test.go

```diff
@@ -0,0 +1,187 @@
+package types
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/tharsis/ethermint/tests"
+)
+
+type IncentiveTestSuite struct {
+	suite.Suite
+}
+
+func TestIncentiveSuite(t *testing.T) {
+	suite.Run(t, new(IncentiveTestSuite))
+}
+
+func (suite *IncentiveTestSuite) TestIncentiveNew() {
+	testCases := []struct {
+		name        string
+		contract    common.Address
+		allocations sdk.DecCoins
+		epochs      uint32
+		expectPass  bool
+	}{
+		{
+			"Register incentive - pass",
+			tests.GenerateAddress(),
+			sdk.DecCoins{sdk.NewDecCoinFromDec("aphoton", sdk.NewDecWithPrec(5, 2))},
+			10,
+			true,
+		},
+		{
+			"Register incentive - empty allocation",
+			tests.GenerateAddress(),
+			sdk.DecCoins{},
+			10,
+			false,
+		},
+		// TODO how to test without panicking?
```

Comment:
@fedekunze can I test the sdk validation without panicking?

Reply:
you can use the literal